test/integration/parameters_permissions_test.rb

```diff
@@ -0,0 +1,139 @@
+require 'integration_test_helper'
+
+class ParametersPermissionsIntegrationTest < ActionDispatch::IntegrationTest
+  setup do
+    role = FactoryGirl.create(:role)
+    @filter = FactoryGirl.create(:filter,
+                                 :permissions => Permission.where(:name => ['view_params']),
+                                 :search => 'name ~ a* or domain_name ~ example*com')
+    domain_filter = FactoryGirl.create(:filter, :permissions => Permission.where(:name => ['edit_domains', 'view_domains']))
+
+    role.filters = [ @filter, domain_filter ]
+    @user = FactoryGirl.create(:user, :mail => 'a@b.com')
```

Comment:
Thanks, I've applied the :with_mail trait, if I use `:roles` it fails on `can_assign?` check unless I configure `User.current` which I want to avoid.